packages/cicero-core/test/template.js

```diff
@@ -28,17 +30,44 @@ chai.use(require('chai-things'));
 chai.use(require('chai-as-promised'));
 
 /* eslint-disable */
+
 function waitForEvent(emitter, eventType) {
     return new Promise((resolve) => {
         emitter.once(eventType, resolve);
     });
 }
 
+function sign(templateHash, timeStamp, path, passPhrase){
+    const p12Ffile = fs.readFileSync(path, { encoding: 'base64' });
+    // decode p12 from base64
+    const p12Der = forge.util.decode64(p12Ffile);
+    // get p12 as ASN.1 object
+    const p12Asn1 = forge.asn1.fromDer(p12Der);
+    // decrypt p12 using the passPhrase 'password'
+    const p12 = forge.pkcs12.pkcs12FromAsn1(p12Asn1, false, passPhrase);
+    //X509 cert forge type
+    const certificateForge = p12.safeContents[0].safeBags[0].cert;
+    //Private Key forge type
+    const privateKeyForge = p12.safeContents[1].safeBags[0].key;
+    //convert cert and private key from forge to PEM
+    const certificatePem = forge.pki.certificateToPem(certificateForge);
+    const privateKeyPem = forge.pki.privateKeyToPem(privateKeyForge);
+    //convert private key in pem to private key type in node
+    const privateKey = crypto.createPrivateKey(privateKeyPem);
+    const sign = crypto.createSign('SHA256');
+    sign.write(templateHash + timeStamp);
+    sign.end();
+    const signature = sign.sign(privateKey, 'hex');
+    return {signature: signature, certificate: certificatePem};
+}
+
 async function writeZip(template){
     try {
         fs.mkdirSync('./test/data/archives');
     } catch (err) {
-        if (err.code !== 'EEXIST') throw err;
+        if (err.code !== 'EEXIST') {
```

Comment:
Did you consider using `fs.existsSync` before `fs.mkdirSync` instead of catching the error here?

Reply:
I haven't written this auxiliary function. Maybe @jeromesimeon might have an idea about this. eslint threw up an error about the code format so I had changed that.

Reply:
hm.... I think this predates even me? But I do believe that I agree with @mttrbrts suggestion.